### PR TITLE
[IMP] Adding filtering by element in the plugins Manager (#18868)

### PR DESCRIPTION
**Pull request by infograf768**
Pull Request for Issue #18868 .

### Summary of Changes
As it is difficult in some languages to Search for a plugin because of the present limitations in Upper/Lower case, this patch adds a new filter by element.
A new pluginelement.php field is created to allow filtering.

Note: took off a useless part of the code concerning filtering by language which certainly came from a wrong copy/paste. 

### Testing Instructions
Display the Plugins Manager

### After patch
screen shot 2017-12-02 at 10 04 23

@coolcat-creations 
@volandku
@mbabker

### CODE REVIEW

File: administrator/components/com_plugins/helpers/plugins.php [hunk 1]
```diff
@@ -100,6 +100,34 @@ public static function folderOptions()
 		return $options;
 	}
 
+	/**
+	 * Returns an array of standard published state filter options.
```
Comment (brianteeman) [inline administrator/components/com_plugins/helpers/plugins.php]: Looks like this is copy pasted and not correct

Comment (infograf768) [reply administrator/components/com_plugins/helpers/plugins.php]: Indeed, same error existed above for folder options.
Would
`Returns an array of folders filter options.`
and
`Returns an array of elements filter options.`
be fine?

Comment (infograf768) [reply administrator/components/com_plugins/helpers/plugins.php]: or rather ` Returns a list of...` etc.

Comment (infograf768) [reply administrator/components/com_plugins/helpers/plugins.php]: will change to a list

**Comment by brianteeman** [comment]
Thanks @infograf768

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 657434a2affc5401da33615bec1bd1f9772e93dd<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18955">issues.joomla.org/tracker/joomla-cms/18955</a>.</sub>

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 657434a2affc5401da33615bec1bd1f9772e93dd<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18955">issues.joomla.org/tracker/joomla-cms/18955</a>.</sub>

**Comment by ghost** [comment]
Ready to Commit after two successful tests.

**Comment by infograf768** [comment]
@mbabker Can it go into 3.8.3 RC ?

**Comment by brianteeman** [comment]
it is a new feature - it is not fixing a bug

**Comment by infograf768** [comment]
Barely a small improvement...

**Comment by brianteeman** [comment]
Semantic versioning doesn't differentiate on impact

**Pull request merged**